packages/eas-cli/src/utils/queries.ts

```diff
@@ -0,0 +1,138 @@
+    promptOptions: {
+      readonly title: string;
+      createDisplayTextForSelectionPromptListItem: (queryItem: QueryReturnType) => string;
+      getIdentifierForQueryItem: (queryItem: QueryReturnType) => string;
+    };
+  };
+
+export async function paginatedQueryWithConfirmPromptAsync<
+  QueryReturnType extends Record<string, any>
+>(
+  {
+    pageSize,
+    offset,
+    queryToPerform,
+    promptOptions,
+  }: PaginatedQueryWithConfirmPromptArgs<QueryReturnType>,
+  accumulator: QueryReturnType[] = []
+): Promise<void> {
+  // query an extra item to determine if there are more pages left
+  const { queryResponse: items, queryResponseRawLength } = await queryToPerform(
+    pageSize + 1,
+    offset
+  );
+  const areMorePagesAvailable = queryResponseRawLength > pageSize;
+  // drop that extra item used for pagination from our render logic when extra pages are available
+  const currentPage = items.slice(0, pageSize);
+  accumulator = [...accumulator, ...currentPage];
+
+  let valueOfUserSelectedListItem = '';
+
+  promptOptions.renderListItems(currentPage);
+  if (areMorePagesAvailable) {
+    valueOfUserSelectedListItem = (await confirmAsync({ message: promptOptions.title }))
+      ? fetchMoreValue
+      : '';
+  }
+
+  if (valueOfUserSelectedListItem === fetchMoreValue) {
+    return await paginatedQueryWithConfirmPromptAsync(
+      {
+        pageSize,
+        offset: offset + pageSize,
+        queryToPerform,
+        promptOptions,
+      },
+      accumulator
+    );
+  }
+}
+
+/**
+ * Returns an array of item(s) where the id is equal to the id of the user's selected item
+ * If no items are available for a user to select, this will return an empty array.
+ */
+export async function paginatedQueryWithSelectPromptAsync<
+  QueryReturnType extends Record<string, any>
+>(
+  {
+    pageSize,
+    offset,
+    queryToPerform,
+    promptOptions,
+  }: PaginatedQueryWithSelectPromptArgs<QueryReturnType>,
+  accumulator: QueryReturnType[] = []
+): Promise<QueryReturnType[]> {
+  // query an extra item to determine if there are more pages left
+  const { queryResponse: items, queryResponseRawLength } = await queryToPerform(
+    pageSize + 1,
+    offset
+  );
+  const areMorePagesAvailable = queryResponseRawLength > pageSize;
+  // drop that extra item used for pagination from our render logic when extra pages are available
+  const currentPage = items.slice(0, pageSize);
+  accumulator = [...accumulator, ...currentPage];
+
+  const selectionPromptListItems = uniqBy(accumulator, queryItem =>
+    promptOptions.getIdentifierForQueryItem(queryItem)
+  ).map(queryItem => ({
+    title: promptOptions.createDisplayTextForSelectionPromptListItem(queryItem),
+    value: promptOptions.getIdentifierForQueryItem(queryItem),
+  }));
+  if (areMorePagesAvailable) {
+    selectionPromptListItems.push({ title: 'Next page...', value: fetchMoreValue });
+  }
+  if (!selectionPromptListItems.length) {
```

Comment:
Don't rely on implicit coercion.  https://www.oreilly.com/library/view/you-dont-know/9781491905159/ch04.html
```suggestion
  if (selectionPromptListItems.length === 0) {
```

Reply:
😱